File: nova-mart-frontend/src/types/promotion.ts
export type PromotionType = 'Percentage' | 'FixedAmount';

export interface Promotion {
  id: number;
  code: string;
  description: string | null;
  type: PromotionType;
  value: number;
  minPurchaseAmount: number;
  maxDiscountAmount: number | null;
  expiresAt: string | null;
  isActive: boolean;
}

export interface CreatePromotionRequest {
  code: string;
  description?: string | null;
  type: PromotionType;
  value: number;
  minPurchaseAmount: number;
  maxDiscountAmount?: number | null;
  expiresAt?: string | null;
}

export interface UpdatePromotionRequest extends CreatePromotionRequest {
  isActive: boolean;
}

export interface PromoValidationResult {
  valid: boolean;
  code: string | null;
  discountAmount: number;
  error: string | null;
}

File: NovaMartPOS.Domain/Entities/Promotion.cs
using NovaMartPOS.Domain.Enums;

namespace NovaMartPOS.Domain.Entities;

public class Promotion : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string? Description { get; set; }
    public PromotionType Type { get; set; }
    public decimal Value { get; set; }
    public decimal MinPurchaseAmount { get; set; } = 0;
    public decimal? MaxDiscountAmount { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public bool IsActive { get; set; } = true;
}